File: EmgenexBusinessPortal/BusinessPanel/src/app/models/sales-rep-model.ts

// export class SalesRepModel {
//     public Id: number
// }

export class SalesRepModel {
    constructor(
        public Id: number = 0,
        public RepGroupName: string = null,
        public SalesDirector: string = null,
        public RepName: string = null,
        public IsActive: boolean = null,
        public OldId: number = null,
        public RepGroupId: number = null,
        public UserId: number = null,
        public SignonDate: DateTimeFormat = null,
        public CreatedBy: number = null,
        public UpdatedBy: number = null,
        public CreatedOn: DateTimeFormat = null,
        public UpdatedOn: DateTimeFormat = null,
        public CreatedByName: string = "",
        public UpdatedByName: string = "",
        public ServiceIds: number[] = [],
        public BusinessId: number = null,
        public SalesRepsComponent: SalesRepsComponent = null,
        public SelectedServiceNames: SelectedServiceNames = null
    ) { }
}

export class SelectedServiceNames {
    constructor(
        public Id: number = null,
        public IsSelected: boolean = false,
        public ParentId: number = null,
        public Selected: string = "",
        public Text: string = "",
        public Value: string = "",
    ) { }
}

export class SalesRepsComponent {
    constructor(
    ) { }
}

export class SalesRepFilter {
    constructor(
        public KeyWords: string = null,
        public pageSize: number = 25,
        public currentPage: number = 1,
        public SortKey: string = null,
        public SortOrder: string = null,
    ) { }
}

File: EmgenexBusinessPortal/Models/SalesRepModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using EBP.Business.Entity.Rep;

namespace EmgenexBusinessPortal.Models
{
    public class SalesRepModel
    {
        public DateTime? CreatedOn = DateTime.UtcNow;

        public DateTime? UpdatedOn = DateTime.UtcNow;

        public int Id { get; set; }

        public int UserId { get; set; }

        public string RepName { get; set; }

        public string RepGroupName { get; set; }

        public int? RepGroupId { get; set; }

        public bool? IsActive { get; set; }

        public DateTime? SignonDate { get; set; }

        public int? CreatedBy { get; set; }

        public int? UpdatedBy { get; set; }

        public int? OldId { get; set; }

        public List<int> ServiceIds { get; set; }

        public string CreatedByName { get; set; }

        public string UpdatedByName { get; set; }
        public List<SelectedItem> SelectedServiceNames { get; set; }
    }
}